tutorials/Tutorial11_Pipelines.py

```diff
@@ -0,0 +1,173 @@
+
+reader = FARMReader(model_name_or_path="deepset/roberta-base-squad2")
+
+######################
+# Prebuilt Pipelines #
+######################
+
+# Extractive QA Pipeline
+########################
+
+p_extractive_premade = ExtractiveQAPipeline(reader=reader, retriever=es_retriever)
+res = p_extractive_premade.run(
+    query="Who is the father of Arya Stark?",
+    top_k_retriever=10,
+    top_k_reader=5
+)
+print_answers(res, details="minimal")
+
+# Document Search Pipeline
+##########################
+
+p_retrieval = DocumentSearchPipeline(es_retriever)
+res = p_retrieval.run(
+    query="Who is the father of Arya Stark?",
+    top_k_retriever=10
+)
+print_documents(res, max_text_len=200)
+
+# Generator Pipeline
+##########################
+
+# We set this to True so that the document store returns document embeddings
+# with each document, this is needed by the Generator
+document_store.return_embedding = True
+
+# Initialize generator
+rag_generator = RAGenerator()
+
+# Generative QA
+p_generator = GenerativeQAPipeline(generator=rag_generator, retriever=dpr_retriever)
+res = p_generator.run(
+    query="Who is the father of Arya Stark?",
+    top_k_retriever=10
+)
+print_answers_gen(res)
+
+# We are setting this to False so that in later pipelines,
+# we get a cleaner printout
+document_store.return_embedding = False
+
+##############################
+# Creating Pipeline Diagrams #
+##############################
+
+p_extractive_premade.draw("pipeline_extractive_premade.png")
+p_retrieval.draw("pipeline_retrievaal.png")
+p_generator.draw("pipeline_generator.png")
+
+####################
+# Custom Pipelines #
+####################
+
+# Extractive QA Pipeline
+########################
+
+# Custom built extractive QA pipeline
+p_extractive = Pipeline()
+p_extractive.add_node(component=es_retriever, name="Retriever", inputs=["Query"])
+p_extractive.add_node(component=reader, name="Reader", inputs=["Retriever"])
+
+# Now we can run it
+res = p_extractive.run(
+    query="Who is the father of Arya Stark?",
+    top_k_retriever=10,
+    top_k_reader=5
+)
+print_answers(res, details="minimal")
+p_extractive.draw("pipeline_extractive.png")
+
+# Ensembled Retriever Pipeline
+##############################
+
+# Create ensembled pipeline
+p_ensemble = Pipeline()
+p_ensemble.add_node(component=es_retriever, name="ESRetriever", inputs=["Query"])
+p_ensemble.add_node(component=dpr_retriever, name="DPRRetriever", inputs=["Query"])
+p_ensemble.add_node(component=JoinDocuments(join_mode="concatenate"), name="JoinResults", inputs=["ESRetriever", "DPRRetriever"])
+p_ensemble.add_node(component=reader, name="Reader", inputs=["JoinResults"])
+p_ensemble.draw("pipeline_ensemble.png")
+
+# Run pipeline
+res = p_ensemble.run(
+    query="Who is the father of Arya Stark?",
+    top_k_retriever=5   #This is top_k per retriever
+)
+print_answers(res, details="minimal")
+
+# Classification Pipeline
```

Comment:
Better name it "Query Classification Pipeline" (could otherwise be confused with ordinary text classification task)